network/src/connectivity_manager/mod.rs

```diff
@@ -75,17 +83,35 @@ pub struct ConnectivityManager<TTicker, TBackoff> {
     event_id: u32,
 }
 
+/// Different sources for peer addresses, ordered by priority (Onchain=highest,
+/// Config=lowest).
+#[repr(u8)]
+#[derive(Copy, Clone, Debug, Eq, PartialEq, Ord, PartialOrd)]
+pub enum DiscoverySource {
+    Onchain = 0,
+    Gossip = 1,
+    Config = 2,
+}
+
+impl DiscoverySource {
+    const NUM_SOURCES: usize = (Self::Config as u8 + 1) as usize;
```

Comment:
This leaves room for programmer error where someone might add an element to the list but forget to update this value. A potential "trick" is to define a final entry in the enum list called "Last", although that would mean you'd need to handle that edge case every time you pattern match on a value of DiscoverySource type.

Reply:
I've just realized there's actually a num-variants crate in common/, so i'll use that!